src/dcqc/utils.py

```diff
@@ -0,0 +1,25 @@
+from fs import open_fs
+from fs.base import FS
+
+
+def validate_from_dict(cls, dictionary):
+    type_ = dictionary.pop("type")
+    if type_ != cls.__name__:
+        message = f"Type ({type_}) does not match the class ({cls.__name__})."
+        raise ValueError(message)
+    return dictionary
+
+
+def open_parent_fs(url: str) -> tuple[FS, str]:
+    # Split off prefix to avoid issues with `rpartition("/")`
+    scheme, separator, path = url.rpartition("://")
+    if separator == "":
+        prefix = "osfs://"
+    else:
+        prefix = scheme + separator
+
+    # parent_path can be "" if there is no "/" in the path
+    parent_path, _, base_name = path.rpartition("/")
+    parent_url = prefix + parent_path
+    fs = open_fs(parent_url)
```

Comment:
Just to confirm if you wanted to download a file: `File("syn://syn50555279").stage()`,

1. This creates a rootless `fs`
2.  rootless fs can actually download a file entity without specifying the full path: `syn://syn50545516/test.txt`

Or am I not understanding this?

Reply:
Resolving this, because this was included as a test case.

Reply:
1. It doesn't necessarily create a rootless FS. It tries to open an FS for the parent directory, and if one cannot be inferred from the URL (_e.g._ `syn://syn50555279`), then it opens a rootless FS. 
2. In the case of a rootless SynapseFS (I can't speak to other FS classes yet), you don't need the full path. You can either refer to the file directly with its Synapse ID, or you can refer to it by name by starting a path with a Synapse ID of one of its parent folders/project and navigating to the file.